### PR TITLE
[flutter_releases] Flutter stable 3.24.0 Framework Cherrypicks

### PR DESCRIPTION
**Pull request by reidbaker**
# Flutter stable 3.24.0 Framework

## Scheduled Cherrypicks
[flutter/153949](https://github.com/flutter/flutter/issues/153949): Fix a crash when deleting text inside CupertinoPageRoute, with a CJK (chinese, japanese, korean) keyboard on Android. 
[flutter/153939](https://github.com/flutter/flutter/issues/153939): Fixes an issue where Flutter TextField may stop accepting input on iOS.
[flutter/152420](https://github.com/flutter/flutter/issues/152420): Fixes scrolling jank when a SelectionArea/SelectableRegion is used as a child of a Scrollable like ListView or PageView on Android and iOS. 
[flutter/154199](https://github.com/flutter/flutter/pull/154199): Removes log spam when building a freshly created template app for Android.
[flutter/153967](https://github.com/flutter/flutter/pull/153967): Fixes macOS host build failure when there are no native asset frameworks to codesign. Impacts users who have enabled experimental native assets feature. 
[flutter/153769](https://github.com/flutter/flutter/pull/153769): When running a Flutter app, display a concise error message when connection to the device is lost.
[flutter/154270](https://github.com/flutter/flutter/pull/154270): Prevent preemptive gradle crash for android builds that would fail to build anyway but with a confusing error message. 
[flutter/54735](https://github.com/flutter/engine/pull/54735): When semantics are enabled on Web, the onTap of various widgets (GestureDetector, InkWell) are called twice or the callback from one widget is called when another widget is pressed.

### CODE REVIEW

**Comment by itsjustkevin** [comment]
Before merging this, please update CHANGELOG.md with the release notes.

**Comment by reidbaker** [comment]
> Before merging this, please update CHANGELOG.md with the release notes.

Done, the tag url does not exist yet but I think that is expected so I guessed it would be 3.24.2

File: CHANGELOG.md [hunk 1]
```diff
@@ -26,6 +26,16 @@ INTERNAL NOTE
 -->
 ## Flutter 3.24 Changes
 
+### [3.24.2](https://github.com/flutter/flutter/releases/tag/3.24.2)
+- [flutter/153949](https://github.com/flutter/flutter/issues/153949): Fix a crash when deleting text inside CupertinoPageRoute, with a CJK (chinese, japanese, korean) keyboard on Android.
```
Comment (itsjustkevin) [inline CHANGELOG.md:30]:
```suggestion
- [flutter/153949](https://github.com/flutter/flutter/issues/153949): Fixes a crash on Android when deleting `EditableText` inside `CupertinoPageRoute`, with a CJK (chinese, japanese, korean) keyboard.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -26,6 +26,16 @@ INTERNAL NOTE
...
+- [flutter/153939](https://github.com/flutter/flutter/issues/153939): Fixes an issue where Flutter TextField may stop accepting input on iOS.
```
Comment (itsjustkevin) [inline CHANGELOG.md:31]:
```suggestion
- [flutter/153939](https://github.com/flutter/flutter/issues/153939): Fixes an issue on iOS where Flutter `TextField`'s may stop accepting input.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -26,6 +26,16 @@ INTERNAL NOTE
...
+- [flutter/152420](https://github.com/flutter/flutter/issues/152420): Fixes scrolling jank when a SelectionArea/SelectableRegion is used as a child of a Scrollable like ListView or PageView on Android and iOS.
```
Comment (itsjustkevin) [inline CHANGELOG.md:32]:
```suggestion
- [flutter/152420](https://github.com/flutter/flutter/issues/152420): Fixes scrolling jank on Android and iOS when a `SelectionArea`/`SelectableRegion` is used as a child of a Scrollable like `ListView` or `PageView`.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -26,6 +26,16 @@ INTERNAL NOTE
...
+- [flutter/154199](https://github.com/flutter/flutter/pull/154199): Removes log spam when building a freshly created template app for Android.
```
[the remark this answers is not in the document]
Comment (itsjustkevin) [reply CHANGELOG.md:33]: I don't understand where this comes in.  Is this when we pass a specific flag.  Why specifically Android?

File: CHANGELOG.md [hunk 5]
```diff
@@ -26,6 +26,16 @@ INTERNAL NOTE
...
+- [flutter/153967](https://github.com/flutter/flutter/pull/153967): Fixes macOS host build failure when there are no native asset frameworks to codesign. Impacts users who have enabled experimental native assets feature.
```
Comment (itsjustkevin) [inline CHANGELOG.md:34]:
```suggestion
- [flutter/153967](https://github.com/flutter/flutter/pull/153967): Fixes a host build failure on macOS when the `native assets` experiment is enabled, and there are no native asset frameworks to codesign.  
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -26,6 +26,16 @@ INTERNAL NOTE
...
+- [flutter/153769](https://github.com/flutter/flutter/pull/153769): When running a Flutter app, display a concise error message when connection to the device is lost.
```
Comment (itsjustkevin) [inline CHANGELOG.md:35]:
```suggestion
- [flutter/153769](https://github.com/flutter/flutter/pull/153769): Simplifies the error message when the connection is lost with a connected device when running a Flutter app.
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -26,6 +26,16 @@ INTERNAL NOTE
...
+- [flutter/154270](https://github.com/flutter/flutter/pull/154270): Prevent preemptive gradle crash for android builds that would fail to build anyway but with a confusing error message.
```
Comment (itsjustkevin) [inline CHANGELOG.md:36]: As a user, this message seems confusing.

File: CHANGELOG.md [hunk 8]
```diff
@@ -26,6 +26,16 @@ INTERNAL NOTE
...
+- [flutter/54735](https://github.com/flutter/engine/pull/54735): When semantics are enabled on Web, the onTap of various widgets (GestureDetector, InkWell) are called twice or the callback from one widget is called when another widget is pressed.
```
Comment (itsjustkevin) [inline CHANGELOG.md:37]:
```suggestion
- [flutter/54735](https://github.com/flutter/engine/pull/54735):  Fixes an error on Flutter Web where `onTap` is called twice on various widgets (`GestureDetector`, `InkWell`) when semantics are enabled.
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -26,6 +26,16 @@ INTERNAL NOTE
 -->
 ## Flutter 3.24 Changes
```
Comment (itsjustkevin) [inline CHANGELOG.md:28]: Were there any Dart changes included in this hotfix?

File: CHANGELOG.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (itsjustkevin) [reply CHANGELOG.md:33]: Thanks for clarifying.

File: CHANGELOG.md (hunk 9 above)
Comment (reidbaker) [reply CHANGELOG.md:28]: https://github.com/dart-lang/sdk/blob/stable/CHANGELOG.md#352---2024-08-28

Comment (reidbaker) [reply CHANGELOG.md:28]: Dart patch notes updated to the format Dart uses.

File: CHANGELOG.md [hunk 10]
```diff
@@ -26,6 +26,29 @@ INTERNAL NOTE
...
+  past the end of the zlib footer (issue [#56481][]).
+- Fixes issue where running `dart` from `PATH` could result in some commands not
+  working as expected (issues [#56080][], [#56306][], [#56499][]).
+- Fixes analysis server plugins not receiving `setContextRoots` requests or
+  being provided incorrect context roots in multi-package workspaces (issue
+  [#56475][]).
+- [flutter/153949](https://github.com/flutter/flutter/issues/153949): Fixes a crash on Android when deleting `EditableText` inside `CupertinoPageRoute`, with a CJK (chinese, japanese, korean) keyboard.
+- [flutter/153939](https://github.com/flutter/flutter/issues/153939): Fixes an issue on iOS where Flutter `TextField`s may stop accepting input.
+- [flutter/152420](https://github.com/flutter/flutter/issues/152420): Fixes scrolling jank on Android and iOS when a `SelectionArea`/`SelectableRegion` is used as a child of a Scrollable like `ListView` or `PageView`.
+- [flutter/154199](https://github.com/flutter/flutter/pull/154199): Removes log spam when building a freshly created template app for Android.
```
Comment (reidbaker) [inline CHANGELOG.md:34]:
```suggestion
- [flutter/154199](https://github.com/flutter/flutter/pull/154199): Removes excessive logging when building a freshly created template app for Android.
```

File: CHANGELOG.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (reidbaker) [reply CHANGELOG.md:34]: Going with "Excessive logging" (thanks go/duckie)

**Review by itsjustkevin — APPROVED** [review]
LGTM

**Pull request merged**